### DISCUSSION TITLE
Sortable list with variable widths

### DISCUSSION THREAD

**Discussion by sebastianservat** [clauderic/dnd-kit#654] [Q&A] [answered]
Hi,

I've built a simple example using SortableContext. I have a row of buttons all with variable widths based on the length of it's label.

When I drag a button from one position to another, I noticed that the width of the placeholders don't change to adapt to the width of the item being dragged. You can see from the screenshots what happens when I drag button `#5` to `#1` position. All other buttons look squashed against one side.

Before dragging
![image](https://user-images.githubusercontent.com/444776/158186897-fa4082fe-5e20-4525-90ef-30d2c3288465.png)

Before dropping
![image](https://user-images.githubusercontent.com/444776/158186995-729d51d9-a7b1-47aa-a737-74f2ef2e1fa1.png)

https://codesandbox.io/s/dnd-kit-sortable-grid-03mt7o

Any ideas why?

Thanks,

Seb.

**Comment by clauderic** [comment]
What [sorting strategy are you using](https://docs.dndkit.com/presets/sortable#sorting-strategies)? Did you try using `horizontalListSortingStrategy`?

**Reply by sebastianservat** [reply]
Thanks for the suggestion @clauderic. I'm using the default `rectSortingStrategy` as I need the sortable items to wrap, thus behave in a grid format. I know you can't tell from the screenshots, but you can resize the table column to the left and the dnd elements would then wrap. 

I tried with `horizontalListSortingStrategy` as you said and the widths of the items behave as expected. That's good. Now, how can I get the `rectSortingStrategy` to behave the same way?

**Reply by sebastianservat** [reply]
Anyone? @clauderic?

Also, another issue I need your help with please:

When using `DragOverlay`, if I drag a <button> and the label is 2 words, the button height increases whilst dragging and the words wraps. Why is this happening?

![image](https://user-images.githubusercontent.com/444776/158527937-0ec36fe3-43cf-4045-8876-f2ecc32490a2.png)

Your help would be greatly appreciated.

Thank you,

Seb.

**Reply by clauderic** [reply]
@sebastianservat it's a bit difficult to help without seeing actual code. Could you share a CodeSandbox replicating the issue you're having?

**Reply by sebastianservat** [reply]
@clauderic ? Really keen to hear your thoughts on the issues I'm facing.

**Comment by sebastianservat** [comment]
@clauderic  - Sounds good, please see below CodeSandbox link:

https://codesandbox.io/s/dnd-kit-sortable-grid-03mt7o 

Thanks,

Seb.

**Comment by clauderic** [comment, marked as answer]
The sorting strategies currently don't work when the layout is not predictable, they either work with grids that have a predictable layout or horizontal/vertical lists (of varying sizes).

Sorting strategies are an optimization that allows you to avoid mutating the DOM while dragging and instead compute the new positions items should be moved to using transforms.

If your layout is not predictable, you don't need to use them. Instead, change the order of the items `onDragOver` instead of `onDragEnd`, and pass in an empty function as the sorting strategy.

https://codesandbox.io/s/dnd-kit-sortable-grid-forked-qkxlc0?file=/src/App.jsx

Refer to this discussion for more information https://github.com/clauderic/dnd-kit/issues/44#issuecomment-757312037

**Reply by sebastianservat** [reply]
@clauderic That is just **genious**! Thank you for the detailed explanation, can't express how happy I am with adopting this library and will gladly sponsor. Great work guys 👏👏👏

**Reply by sebastianservat** [reply]
@clauderic  I'm trying to extend the CodeSandbox example you helped me with to allow dragging of elements to another container. I've been trying to port the logic from your StoryBook MultipleContainers.tsx, but I'm finding it hard to follow.

If it's not too much hassle, would you mind adding an example on the previous CodeSandbox? Otherwise, perhaps in general terms guide me through the minimum setup required in order to allow dragging between containers?

Thanks,

Seb.
